homeassistant/components/switch_as_x/__init__.py

```diff
@@ -20,9 +20,67 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+@callback
+def async_add_to_device(
+    hass: HomeAssistant, entry: ConfigEntry, entity_id: str
+) -> str | None:
+    """Add our config entry to the tracked entity's device."""
+    registry = er.async_get(hass)
+    device_registry = dr.async_get(hass)
+    device_id = None
+
+    wrapped_switch = registry.async_get(entity_id)
+    if (
+        not (wrapped_switch := registry.async_get(entity_id))
+        or not (device_id := wrapped_switch.device_id)
+        or not (device_entry := device_registry.async_get(device_id))
+    ):
+        return device_id
+
+    device_registry.async_update_device(device_id, add_config_entry_id=entry.entry_id)
+
+    if (
+        not wrapped_switch.config_entry_id
+        or wrapped_switch.config_entry_id not in device_entry.config_entries
+    ):
+        return device_id
+
+    @callback
+    def async_device_updated(event):
+        """Handle the update of a device.
+
+        Will remove our config entry from the switch's device if the switch's config
```

Comment:
I wonder if we already cover this case with the entity registry event listener below?

Reply:
You're right, that's handled here:

https://github.com/home-assistant/core/blob/66d757115cdb49b728b4e270a2b8359422a374c1/homeassistant/helpers/entity_registry.py#L460-L470